src/api/routers/topic.ts

```diff
@@ -41,18 +50,40 @@ export const topicRouter = router({
     .input(
       z.object({
         username: userSchema.shape.username,
-        title: topicSchema.shape.title,
+        title: z.string(),
       }),
     )
     .query(async (opts) => {
       const isCreator = opts.input.username === opts.ctx.user?.username;
+      const normalizedTitle = normalizeTitle(decodeURIComponent(opts.input.title));
+      const topics = await xprisma.$queryRaw<
+        {
+          id: number;
+          title: string;
+          creatorName: string;
+          description: string;
+          visibility: string;
+          allowAnyoneToEdit: boolean;
+          createdAt: Date;
+          updatedAt: Date;
+        }[]
+      >`
+        SELECT id, title, "creatorName", description, visibility, "allowAnyoneToEdit", "createdAt", "updatedAt"
+        FROM "topics"
+        WHERE "creatorName" = ${opts.input.username}
+        AND (
+          REPLACE(LOWER(title), ' ', '-') = ${normalizedTitle}
+          OR title = ${normalizedTitle}
+        )
+        AND (${isCreator} OR visibility != 'private')
+        LIMIT 1
+      `;
+
+      if (!topics[0]) return null;
 
+      // Fetch related data since raw query doesn't support include -- Added to match original behavior
       return await xprisma.topic.findFirst({
```

Comment:
`findUnique` is clearer/I think slightly more performant if we're querying by id.
```suggestion
      return await xprisma.topic.findUnique({
```